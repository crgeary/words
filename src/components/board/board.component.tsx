import { FC } from "react";
import { getGuessedRowState } from "../../utils/get-guessed-row-state";
import { Row } from "../row/row.component";

export type BoardProps = {
    solution: string;
    rows: number;
    guesses: string[];
    currentGuess: string;
};

export const Board: FC<BoardProps> = ({ solution, rows, guesses, currentGuess }) => {
    const board = Array.from({
        ...[
            ...guesses.map((letters) => {
                const states = getGuessedRowState(letters, solution);
                return letters.split("").map((char, i) => ({ children: char, state: states[i] }));
            }),
            currentGuess.split("").map((char) => ({ children: char })),
        ],
        length: rows,
    });
    return (
        <div className="h-full grid auto-rows-fr gap-1">
            {board.map((letters, i) => (
                <Row key={i} letters={letters} />
            ))}
        </div>
    );
};
