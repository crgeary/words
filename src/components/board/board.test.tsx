import { render } from "@testing-library/react";
import { Board } from "./board.component";

test("it should match snapshot", () => {
    const { container } = render(
        <Board rows={5} solution="heros" guesses={["hello", "world"]} currentGuess="" />,
    );
    expect(container.firstChild).toMatchSnapshot();
});
