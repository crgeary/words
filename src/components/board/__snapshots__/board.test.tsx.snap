// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match snapshot 1`] = `
<div
  class="h-full grid auto-rows-fr gap-1"
>
  <div
    class="grid grid-cols-5 gap-1"
  >
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-green-500 border-green-600 text-green-50"
    >
      <span
        class="uppercase font-medium text-xl"
      >
        h
      </span>
    </span>
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-green-500 border-green-600 text-green-50"
    >
      <span
        class="uppercase font-medium text-xl"
      >
        e
      </span>
    </span>
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-slate-500 border-slate-600 text-slate-200"
    >
      <span
        class="uppercase font-medium text-xl"
      >
        l
      </span>
    </span>
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-slate-500 border-slate-600 text-slate-200"
    >
      <span
        class="uppercase font-medium text-xl"
      >
        l
      </span>
    </span>
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-yellow-500 border-yellow-600 text-yellow-900"
    >
      <span
        class="uppercase font-medium text-xl"
      >
        o
      </span>
    </span>
  </div>
  <div
    class="grid grid-cols-5 gap-1"
  >
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-slate-500 border-slate-600 text-slate-200"
    >
      <span
        class="uppercase font-medium text-xl"
      >
        w
      </span>
    </span>
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-yellow-500 border-yellow-600 text-yellow-900"
    >
      <span
        class="uppercase font-medium text-xl"
      >
        o
      </span>
    </span>
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-green-500 border-green-600 text-green-50"
    >
      <span
        class="uppercase font-medium text-xl"
      >
        r
      </span>
    </span>
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-slate-500 border-slate-600 text-slate-200"
    >
      <span
        class="uppercase font-medium text-xl"
      >
        l
      </span>
    </span>
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-slate-500 border-slate-600 text-slate-200"
    >
      <span
        class="uppercase font-medium text-xl"
      >
        d
      </span>
    </span>
  </div>
  <div
    class="grid grid-cols-5 gap-1"
  >
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-white border-violet-200 text-slate-600"
    />
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-white border-violet-200 text-slate-600"
    />
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-white border-violet-200 text-slate-600"
    />
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-white border-violet-200 text-slate-600"
    />
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-white border-violet-200 text-slate-600"
    />
  </div>
  <div
    class="grid grid-cols-5 gap-1"
  >
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-white border-violet-200 text-slate-600"
    />
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-white border-violet-200 text-slate-600"
    />
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-white border-violet-200 text-slate-600"
    />
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-white border-violet-200 text-slate-600"
    />
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-white border-violet-200 text-slate-600"
    />
  </div>
  <div
    class="grid grid-cols-5 gap-1"
  >
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-white border-violet-200 text-slate-600"
    />
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-white border-violet-200 text-slate-600"
    />
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-white border-violet-200 text-slate-600"
    />
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-white border-violet-200 text-slate-600"
    />
    <span
      class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-white border-violet-200 text-slate-600"
    />
  </div>
</div>
`;
